### ISSUE TITLE
Encompass data tab-panel and subcategories side panel under one heading

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#841]
**Page 61 in DAC report**

**Issue from DAC report:**
The purpose of the Sub-category components is visually indicated by the positioning of the section alongside the tab-panel containing the data.

The h4 heading ‘Sub-categories’ is one level below the h3 heading within the tab panel, but as this had been implemented above the tab panel within the HTML, the headings are illogical and the hierarchical relationship is not properly defined. This means it is not clear how the ‘Sub-categories’ panel is related to the tab-panel - see hierarchies image.

**Solution from DAC report:**
Consider implementing a heading to encompass both the tab-panel and the Sub-category options. The ‘Sub-categories’ heading could then still be implemented as a `<h4>`. ‘Sex’ and ‘Age’ could be a h5 or alternatively removed as they are not descriptive. To provide extra context to the ‘Sub-categories’ heading, it could be changed to ‘Choose data sub-categories to view’.

**Example from DAC report:**

```
<div class="row no-units" id="indicatorData" data-indicatorid="indicator_1-2-1"
data-id="1-2-1" data-country="United Kingdom" data-charttitle="Percentage of the
population living in households at risk of poverty" data-charttitles="null"
data-measurementunit="Percentage (%)" data-datasource="Eurostat" datacopyright=""
data-geographicalarea="United Kingdom" data-footnote="" datashowdata="
true" data-graphtype="line" data-geocoderegex="" data-showmap="false"
data-startvalues="null" data-graphlimits="null" data-stackeddisaggregation="">
<h4>View data</h4>
…
</div>
```

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin This one is hard for me to understand. I suspect it's because our heading structure is not particularly consistent, and also we have some optional things that don't always appear. To help me understand the current situation, here's a pseudo-code representation of our indicator page right now. The `optional` class is intended to show that some things do not always appear.

```
<h1>Indicator number</h1>
<p>Indicator name</p>
<h2 class="optional">Indicator available</h2>
<sidebar-section>
  <h4 class="optional">Series</h4>
  <h4 class="optional">Units</h4>
  <h5 class="optional">Subcategories</h5>
</sidebar-section>
<main-section>
  <chart-tab>
    <h3>Chart title</h3>
  </chart-tab>
  <table-tab>
    <caption>Chart title</caption>
  </table-tab>
</main-section>
```

Maybe we could come up with an ideal structure by changing this pseudo-code?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I've read over it again and I'm also not quite sure.

This image of the current hierarchy was included in the report:
![image](https://user-images.githubusercontent.com/43404358/91882357-c2684500-ec7a-11ea-9a91-b6b254caaf2a.png)

I think they might be suggesting it be like this instead:
![image](https://user-images.githubusercontent.com/43404358/91882945-9f8a6080-ec7b-11ea-9c43-558862b077c3.png)

Looking at your pseudo code now to see how it translates into that?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning actually I guess it kind of differs depending on whether or not there's series and units as looking a the pseudo-code they're h4 and subcategores is h5

**Comment by brockfanning** [comment]
Quick note - I realized my pseudo-code was wrong in one place, so I edited it. The "Subcategories" should be an `H4` and the individual fields (which I forgot to add) are `H5`.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning ok, thanks - I think a big part of this being tricky is the optional indicator available field.

I'm not sure what you think or what DAC thinks (anything we come up with today, I can run past them in the morning), but maybe this:

Added a `h3` in for the sub category, as at least then as long as indicator available is present the sidebar and chart (chart-title) are encompassed under one heading (`h2`), and then in docs, could say something like for hierarchy to be correct ensure indicator available is populated?

```
<h1>Indicator number</h1>
<p>Indicator name</p>
<h2 class="optional">Indicator available</h2>
<h3>Data selection</h3>
<sidebar-section>
  <h4 class="optional">Series</h4>
  <h4 class="optional">Units</h4>
  <h4 class="optional">Subcategories</h4>
  <h5 class="optional">Sex</h5>
  <h5 class="optional">Age</h5>
  <h5 class="optional">etc...</h5>
</sidebar-section>
<main-section>
  <chart-tab>
    <h3>Chart title</h3>
  </chart-tab>
  <table-tab>
    <caption>Chart title</caption>
  </table-tab>
</main-section>
```

**Comment by brockfanning** [comment]
To throw out an alternative, what if make the chart title a `<p>` instead of a heading? From what I understand, the problem is that we have an `<h3>` after an `<h4>`. So, if we didn't have that `<h3>` at all, is the problem solved?

Also a note - I'm not sure whether it's a problem to "skip" heading levels. According to this page, it is not technically a WCAG failure to skip heading levels: https://developer.paciellogroup.com/blog/2020/03/heading-off-confusion-when-do-headings-fail-wcag/

**Comment by LucyGwilliamAdmin** [comment]
That would also be a good alternative - is there a way to get it the same size as the `<p>` we use for the indicator name?

**Comment by brockfanning** [comment]
Sure, I think we should be able to use that same "lead-copy" class.

**Comment by LucyGwilliamAdmin** [comment]
Ok, great - I'll make sure we bring that up with DAC tomorrow then, and will also ask about skipping headings just so we get their thoughts on it aswell

**Comment by LucyGwilliamAdmin** [comment]
@phillipgardiner @brockfanning response from DAC on this:

1.	In terms of the headings structure, headings need to be logical and hierarchical as the levels present to screen reader users the relationships within the page. This means that headings should not skip a level. For example, it doesn’t make sense to go from a <h2> straight to a <h4> because a <h4> element implies that it is a sub-heading of a particular level (i.e. 3 levels below the main heading). But if there is no <h3> to which it belongs then this is confusing. The example you have given for 5.2.1 is also an illogical heading structure I am afraid as it skips from a <h2> to a <h4>. However, I would say that illogical headings are only a (AAA) issue, and that the reason this was failed so highly in the report (A) was because the association between the Sub-categories and the charts was not clear, and implementing a correct heading structure would correct this. On the call, Andy clarified that the heading level 2 ‘Percentage of the population…’ was sufficient to create this association, especially if you add some text such as ‘data’ to provide some extra context. So while we would advise that the (AAA) standard were met by correcting the hierarchical structure, this is not necessary for (A) standard.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning slightly unrelated but another thing that came up in call was about indented dropdowns being lower down in hierarchy e.g on [UK 3.b.1](https://sdgdata.gov.uk/3-b-1/), Vaccination by age is h5, so Vaccine and Region should be h6. Let me know if you think that should be separate ticket?

Also I'm not sure if we have any h6s already in platform? Would we want them styled the same as h5s? @phillipgardiner what are your thoughts?

**Comment by phillipgardiner** [comment]
@brockfanning @LucyGwilliamAdmin Sorry still trying to get me head around this! Does the header structure e.g. H1 given affect the look and feel? I believe they are two separate things as styling can be done via CSS despite the header number given. Therefore we should for example be able to have a H4 that is visually bigger than a H1 despite is being further down the header structure. Is this correct? If so I think its important we differentiate between the heading structure and the style associated with that header.

**Comment by LucyGwilliamAdmin** [comment]
@phillipgardiner I think as long as hierarchy is correct it's fine. In they're reply the main thing they said to be careful of was skipping heading as it doesn't pass AAA standard but we're only going for AA if we want/need to we can skip

**Comment by brockfanning** [comment]
The idea of making the heading level reflected in the disaggregation hierarchy (ie, a parent disaggregation is h5, and child disaggregation h6, etc.) should be another ticket, I think. It does make sense. It may be technically tricky but that shouldn't stop us from trying, either with headings or something like `aria-level` attributes.

Meanwhile I'll go ahead with changing the chart title to an `h4`, which I believe is what we've settled on here?
